tests/trainer/test_trainer.py

```diff
@@ -808,6 +810,7 @@ def training_step(self, batch, batch_idx):
     with pytest.raises(ValueError, match=r".*The loss returned in `training_step` is.*"):
         trainer.fit(model)
         assert trainer.global_step == model.test_batch_inf
+        assert backward_mock.call_count == model.test_batch_inf
```

Comment:
reviewer: the test here ensures we are terminating before the 4th backward pass.